public/scripts/d3/d3.js

```diff
@@ -1 +1,130 @@
-//  Add d3 functions here
+d3.json(echo.margins.jsonUrl, (err, data) => {
+  data.sort((a, b) => d3.descending(a.avgRating, b.avgRating));
+
+  let uniqueTags = [];
+  data.forEach((d) => {
+    uniqueTags.push(d.tag);
+  });
+  uniqueTags = uniqueTags.filter(onlyUnique);
+
+  const nodePosition = [];
+  const startingCx = 160;
+  let startingCy = 120;
+  const sortedByTag = {};
+
+  uniqueTags.forEach((t) => {
```

Comment:
Hi Eoin, this is definitely much cleaner but we can't seem to get it to work? [Here's](https://repl.it/JHCR) a repl that we tried, however it never runs past the first node, after the first round sortedByTag is just coming back as undefined..